utility/resource_loader_compat.cpp

```diff
@@ -0,0 +1,2096 @@
+	Error error = OK;
+	float prog;
+	auto loader = _open(p_path, "", true, &error, &prog);
+	ERR_RFLBC_COND_V_MSG_CLEANUP(error != OK, error, "Cannot open resource '" + p_path + "'.", loader);
+	ERR_RFLBC_COND_V_MSG_CLEANUP(loader->engine_ver_major != 2, ERR_INVALID_DATA, "Not a V2 Resource: '" + p_path + "'.", loader);
+	// In case this is a imageTexture resource, we have to set this
+	loader->convert_v2image_indexed = true;
+	loader->hacks_for_deprecated_v2img_formats = false;
+
+	error = loader->load();
+	ERR_RFLBC_COND_V_MSG_CLEANUP(error != OK, error, "Cannot load resource '" + p_path + "'.", loader);
+	loader->imd = imd;
+	error = loader->save_to_bin(p_dst);
+	memdelete(loader);
+	ERR_FAIL_COND_V_MSG(error != OK, error, "Cannot save resource to '" + p_dst + "'.");
+	return error;
+}
+
+Error ResourceFormatLoaderCompat::get_import_info(const String &p_path, const String &base_dir, Ref<ImportInfo> &i_info) {
+	Error error = OK;
+
+	ResourceLoaderBinaryCompat * loader = _open(p_path, base_dir, true, &error, nullptr);
+	ERR_RFLBC_COND_V_MSG_CLEANUP(error != OK, error, "failed to open resource '" + p_path + "'.", loader);
+
+	if (i_info == nullptr){
+		i_info.instance();
+	}
+
+	i_info->import_path = loader->local_path;
+	i_info->type = loader->type;
+	i_info->version = loader->engine_ver_major;
+	if (loader->engine_ver_major == 2){
+		i_info->dest_files.push_back(loader->local_path);
+		i_info->import_md_path = loader->res_path;
+		//these do not have any metadata info in them
+		if(i_info->import_path.find(".converted.") != -1)
+		{
+			memdelete(loader);
+			return OK;
+		}
+		Error error = loader->load_import_metadata();
+		ERR_RFLBC_COND_V_MSG_CLEANUP(error != OK, ERR_PRINTER_ON_FIRE, "failed to get metadata for '" + loader->res_path + "'",loader);
+
+		if (loader->imd->get_source_count() > 1){
+			WARN_PRINT("More than one source?!?!");
+		}
+		ERR_RFLBC_COND_V_MSG_CLEANUP(loader->imd->get_source_count() == 0, ERR_FILE_CORRUPT, "metadata corrupt for '" + loader->res_path + "'", loader);
+		i_info->v2metadata = loader->imd;
+		i_info->source_file = loader->imd->get_source_path(0);
+		i_info->importer = loader->imd->get_editor();
+		i_info->params = loader->imd->get_options_as_dictionary();
+	}
+	memdelete(loader);
+	return OK;
+}
+
+
+ResourceLoaderBinaryCompat * ResourceFormatLoaderCompat::_open(const String &p_path, const String &base_dir, bool fake_load, Error *r_error, float *r_progress){
+	Error error = OK;
+	if (!r_error) {
+		r_error = &error;
+	}
+	String res_path = GDRESettings::get_singleton()->get_res_path(p_path, base_dir);
+	FileAccess *f = nullptr;
+	if (res_path != ""){
+		f = FileAccess::open(res_path, FileAccess::READ, r_error);
+	} else {
+		*r_error = ERR_FILE_NOT_FOUND;
+		ERR_FAIL_COND_V_MSG(!f, nullptr, "Cannot open file '" + res_path + "'.");
+	}
+	// TODO: remove this extra check
+	ERR_FAIL_COND_V_MSG(!f, nullptr, "Cannot open file '" + res_path + "' (Even after get_res_path() returned a path????).");
```

Comment:
Do not use multiple `?` in the row, it can be incorrectly interpreted as [trigraph](https://docs.microsoft.com/en-us/cpp/c-language/trigraphs?view=msvc-160).

```suggestion
	ERR_FAIL_COND_V_MSG(!f, nullptr, "Cannot open file '" + res_path + "' (Even after get_res_path() returned a path?).");
```